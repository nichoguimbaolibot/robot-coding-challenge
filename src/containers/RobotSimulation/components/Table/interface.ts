export interface ITable {
  rowBoxes?: number;
}

export interface ICoordinates {
  rowBoxes: number;
  block?: boolean;
}
