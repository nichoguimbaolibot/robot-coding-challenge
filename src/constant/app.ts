export const MAX_BOXES = 5;
