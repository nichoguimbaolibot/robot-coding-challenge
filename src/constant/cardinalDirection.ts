export const NORTH = 'North';
export const EAST = 'East';
export const SOUTH = 'South';
export const WEST = 'West';

export const CARDINAL_DIRECTION = [
  {
    id: 1,
    direction: 0,
    position: NORTH,
  },
  {
    id: 2,
    direction: 90,
    position: EAST,
  },
  {
    id: 3,
    direction: 180,
    position: SOUTH,
  },
  {
    id: 4,
    direction: 270,
    position: WEST,
  },
];

export interface ICardinalDirection {
  direction?: number;
  id?: number;
  position?: string;
}
