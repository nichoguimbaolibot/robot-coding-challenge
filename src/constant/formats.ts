export const DATE_FORMAT = 'LTS';
