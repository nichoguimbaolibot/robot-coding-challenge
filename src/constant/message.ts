const MESSAGES = {
  INVALID_POSITION: 'Buddy~ It was an invalid position.',
  MOVEMENT_NOT_ALLOWED: 'Hey!! I will fall off the table!!',
  ROBOT_NOT_FOUND: 'Im not in the table broooo~',
};

export default MESSAGES;
