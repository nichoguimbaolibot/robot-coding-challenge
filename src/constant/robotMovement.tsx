export const ROBOT_MOVEMENT = {
  DEFAULT_X_AXIS_POSITION: 49,
  DEFAULT_Y_AXIS_POSITION: 420,
  DEFAULT_X_AXIS_MOVEMENT: 96,
  DEFAULT_Y_AXIS_MOVEMENT: 103,
};
