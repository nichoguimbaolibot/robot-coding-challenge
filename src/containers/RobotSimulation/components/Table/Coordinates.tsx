import React from 'react';
import classnames from 'classnames';
import { ICoordinates } from './interface';

const Coordinates: React.FunctionComponent<ICoordinates> = ({
  rowBoxes,
  block,
}): JSX.Element => {
  const coordinatesClassName = classnames('coordinate-section', {
    '-block': block,
    '-inline': !block,
  });
  return (
    <section className={coordinatesClassName}>
      {Array(rowBoxes)
        .fill(Math.random())
        .map((box: number, index: number) => (
          <span className="coordinate-label" key={index}>
            {index}
          </span>
        ))}
    </section>
  );
};

export default Coordinates;
