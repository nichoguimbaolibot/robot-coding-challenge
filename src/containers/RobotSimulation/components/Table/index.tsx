import React, { FunctionComponent } from 'react';
import Box from '../Box';
import Coordinates from './Coordinates';
import { ITable } from './interface';

const Table: FunctionComponent<ITable> = ({ rowBoxes }): JSX.Element => {
  const boxes = rowBoxes ? rowBoxes : 5;

  return (
    <div className="table-box-section">
      <div className="table-item">
        <div className="vertical-coordinates">
          <Coordinates rowBoxes={boxes} block />
        </div>
        <div className="box-generator">
          {Array(boxes)
            .fill(Math.random())
            .map((box: number, index: number) => (
              <Box key={index} rowBoxes={boxes} />
            ))}
          <Coordinates rowBoxes={boxes} />
        </div>
      </div>
    </div>
  );
};
export default Table;
