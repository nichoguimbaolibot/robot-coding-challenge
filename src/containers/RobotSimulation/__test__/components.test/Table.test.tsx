import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Table from 'containers/RobotSimulation/components/Table';
import { MAX_BOXES } from 'constant/app';

afterEach(cleanup);

describe('test TABLE component', () => {
  it('renders TABLE component', () => {
    const { asFragment } = render(<Table rowBoxes={MAX_BOXES} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
