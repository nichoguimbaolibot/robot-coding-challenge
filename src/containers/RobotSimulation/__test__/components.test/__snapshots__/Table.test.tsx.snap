// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test TABLE component renders TABLE component 1`] = `
<DocumentFragment>
  <div
    class="table-box-section"
  >
    <div
      class="table-item"
    >
      <div
        class="vertical-coordinates"
      >
        <section
          class="coordinate-section -block"
        >
          <span
            class="coordinate-label"
          >
            0
          </span>
          <span
            class="coordinate-label"
          >
            1
          </span>
          <span
            class="coordinate-label"
          >
            2
          </span>
          <span
            class="coordinate-label"
          >
            3
          </span>
          <span
            class="coordinate-label"
          >
            4
          </span>
        </section>
      </div>
      <div
        class="box-generator"
      >
        <div
          class="box-container"
          data-testid="box-container-test"
        >
          <div
            class="box"
            data-testid="box-test"
          />
          <div
            class="box"
            data-testid="box-test"
          />
          <div
            class="box"
            data-testid="box-test"
          />
          <div
            class="box"
            data-testid="box-test"
          />
          <div
            class="box"
            data-testid="box-test"
          />
        </div>
        <div
          class="box-container"
          data-testid="box-container-test"
        >
          <div
            class="box"
            data-testid="box-test"
          />
          <div
            class="box"
            data-testid="box-test"
          />
          <div
            class="box"
            data-testid="box-test"
          />
          <div
            class="box"
            data-testid="box-test"
          />
          <div
            class="box"
            data-testid="box-test"
          />
        </div>
        <div
          class="box-container"
          data-testid="box-container-test"
        >
          <div
            class="box"
            data-testid="box-test"
          />
          <div
            class="box"
            data-testid="box-test"
          />
          <div
            class="box"
            data-testid="box-test"
          />
          <div
            class="box"
            data-testid="box-test"
          />
          <div
            class="box"
            data-testid="box-test"
          />
        </div>
        <div
          class="box-container"
          data-testid="box-container-test"
        >
          <div
            class="box"
            data-testid="box-test"
          />
          <div
            class="box"
            data-testid="box-test"
          />
          <div
            class="box"
            data-testid="box-test"
          />
          <div
            class="box"
            data-testid="box-test"
          />
          <div
            class="box"
            data-testid="box-test"
          />
        </div>
        <div
          class="box-container"
          data-testid="box-container-test"
        >
          <div
            class="box"
            data-testid="box-test"
          />
          <div
            class="box"
            data-testid="box-test"
          />
          <div
            class="box"
            data-testid="box-test"
          />
          <div
            class="box"
            data-testid="box-test"
          />
          <div
            class="box"
            data-testid="box-test"
          />
        </div>
        <section
          class="coordinate-section -inline"
        >
          <span
            class="coordinate-label"
          >
            0
          </span>
          <span
            class="coordinate-label"
          >
            1
          </span>
          <span
            class="coordinate-label"
          >
            2
          </span>
          <span
            class="coordinate-label"
          >
            3
          </span>
          <span
            class="coordinate-label"
          >
            4
          </span>
        </section>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
